### PR TITLE
Orekit validation first approach

### PR DESCRIPTION
**Pull request by jorgepiloto**
This pull request is a first approach to poliastro's validation path. It makes use of the Orekit python wrapper and implements the following features:

* A class under the name `OrekitOrbit`, who's usage is similar to poliastro's `Orbit` class. Both of them try to model Keplerian orbits, which can be defined from vectors or classic orbital elements.

* A simple test regarding the conversion between RV and COE. 

:warning: Orekit does not provide functions like `rv2coe` or `coe2rv`, since conversion is done directly in each of the available constructors of the `KeplerianOrbit` class in the Orekit library :warning: 

Also notice the addition of the `orekit-data.zip` file, which holds necessary data for Orekit to properly work: ephemeris, time files... We might discuss if this data should be included within this repo or final user is required to download it from official channels.

### CODE REVIEW

**Comment by astrojuanlu** [comment]
> We might discuss if this data should be included within this repo or final user is required to download it from official channels.

Let's not put it in the repo :pray: Otherwise it will make the whole git history too sluggish. If it's a common Orekit thing we can point to the relevant documentation, or even provide a script to download it.

I'll take a look at this tomorrow!

**Comment by jorgepiloto** [comment]
I just saw that the Orekit wrapper provides a [download data function](https://gitlab.orekit.org/orekit-labs/python-wrapper/-/blob/master/python_files/pyhelpers.py#L40).

On the other hand, if the `setup_orekit_dir()` does not find the `orekit-data.zip` file in the local directory, en error message is shown but **no exception is raised**. This is weird when trying to manage program workflow...

**Comment by jorgepiloto** [comment]
**Some updates on this PR regarding previous suggestions and feedback**

I came up with the following structure:

```bash
orekit/
├── README.md
├── requirements.txt
├── src/
└── tests/
```

The `src/` directory holds all logical files while the `tests/` one is devoted to validation unit tests. In addition, a `pytest.ini` and `conftest.py` files are placed under this last folder, so anyone can run the tests following the same configuration. 

Regarding `orekit-data.zip`:

> Let's not put it in the repo pray Otherwise it will make the whole git history too sluggish. If it's a common Orekit thing we can point to the relevant documentation, or even provide a script to download it.

Now, there exists a `setup_orekit_env()` function under `src/utils.py` which checks if data is available in local machine. If not, it automatically downloads it from official sources and points `orekit` internals to data location, which I selected to be within `src/data/`.

File: orekit/README.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+wrapper](https://gitlab.orekit.org/orekit-labs/python-wrapper).
+
+
+### How to install
+
+A `requirements.txt` file is provided. Preferred way of installation, due to the
+nature of Orekit, is through `conda` by running:
+
+```bash
+conda install --file requirements.txt
```
[the remark this answers is not in the document]
Comment (astrojuanlu) [reply orekit/README.md:14]: Regardless, these instructions are incomplete:

```
$ conda install --file requirements.txt 
Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Collecting package metadata (repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.

PackagesNotFoundError: The following packages are not available from current channels:

  - orekit

Current channels:

  - https://repo.anaconda.com/pkgs/main/linux-64
  - https://repo.anaconda.com/pkgs/main/noarch
  - https://repo.anaconda.com/pkgs/r/linux-64
  - https://repo.anaconda.com/pkgs/r/noarch

To search for alternate channels that may provide the conda package you're
looking for, navigate to

    https://anaconda.org

and use the search bar at the top of the page.
```

[the remark this answers is not in the document]
Comment (astrojuanlu) [reply orekit/README.md:14]: And also, poliastro is missing as a dependency :)

File: orekit/tests/test_twobody_orbit.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+    # Generate orekit and poliastro orbits
+    ss_orekit = OrekitOrbit.from_vectors(Earth, r, v)
+    ss_poliastro = Orbit.from_vectors(Earth, r, v)
+
+    # Assert position and velocity vectors
+    assert_quantity_allclose(ss_poliastro.a, ss_orekit.a)
+    assert_quantity_allclose(ss_poliastro.ecc, ss_orekit.ecc)
+    assert_quantity_allclose(ss_poliastro.inc, ss_orekit.inc)
+    assert_quantity_allclose(ss_poliastro.raan, ss_orekit.raan)
```
Comment (astrojuanlu) [inline orekit/tests/test_twobody_orbit.py:31]: This test is currently failing on my laptop:

```
E       AssertionError: 
E       Not equal to tolerance rtol=1e-07, atol=0
E       
E       Mismatched elements: 1 / 1 (100%)
E       Max absolute difference: 6.28318531
E       Max relative difference: 3.43771527
E        x: array(4.455464)
E        y: array(-1.827721)
```

**Review by astrojuanlu — CHANGES_REQUESTED** [review]
This is on the right path! Left a few comments.

File: README.md [hunk 3]
```diff
@@ -1,5 +1,28 @@
-# poliastro validation
+poliastro validation
```
[the remark this answers is not in the document]
Comment (jorgepiloto) [reply README.md:1]: Sure!

File: orekit/README.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
+Orekit validation framework
+===========================
+
+The following directory holds a validation framework against the [Orekit Python
+wrapper](https://gitlab.orekit.org/orekit-labs/python-wrapper).
+
+
+### How to install
```
[the remark this answers is not in the document]
Comment (jorgepiloto) [reply orekit/README.md:8]: Sorry for this mess with the Markdown syntax... I do not like the size of second level headings when displayed in the GitHub repositories :laughing: 

Solving this immediately :+1:

File: orekit/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (jorgepiloto) [reply orekit/README.md:14]: I just tried the `environment.yml` file and this solved for previous issue of `PackagesNotFoundError` by adding `conda-forge` within the `channels` category. The usage of this YAML file seems the best approach for resolving dependencies issues.

Finally, I also added poliastro to the deps collections as you reminded me.

File: orekit/src/utils.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    # Official data download link
+    url = "https://gitlab.orekit.org/orekit/orekit-data/-/archive/master/orekit-data-master.zip"
+
+    # Start downloading process
+    r = requests.get(url, stream=True)
+    with open(save_path, "wb") as fd:
+        print("Downloading data...")
+        for chunk in r.iter_content(chunk_size=chunk_size):
+            fd.write(chunk)
+    print("[OK]")
```
[the remark this answers is not in the document]
Comment (jorgepiloto) [reply orekit/src/utils.py]: I am not familiar with this module and thus, instantiated the logger inside the function. Is this correct? Not sure if the object should have a global scope...

File: orekit/tests/test_twobody_orbit.py (hunk 2 above)
Comment (jorgepiloto) [reply orekit/tests/test_twobody_orbit.py:32]: Orekit computes the `raan` in the following [line of code](https://gitlab.orekit.org/orekit/orekit/-/blob/develop/src/main/java/org/orekit/orbits/KeplerianOrbit.java#L321):

```java
raan = Vector3D.crossProduct(Vector3D.PLUS_K, momentum).getAlpha();
```

Previous `getAlpha()` bounds the output angle between `-pi` and `+pi`, as stated in Hipparchus (the Java math library I guess) [official docs](https://www.hipparchus.org/apidocs/org/hipparchus/geometry/euclidean/threed/Vector3D.html#getAlpha--). However, this angle is claimed to be within the range `[0, 2pi)` in Vallado book and the approach followed in poliastro `rv2coe`.

I just bounded the angle between 0 and 2pi and all the tests are passing now :+1:

**Comment by jorgepiloto** [comment]
Sorry, I just forgot to update the README.md installation section guide. I will solve for this this night.

**Comment by jorgepiloto** [comment]
I just added a `propagate()` method and its corresponding validation test, which passes without any problem. I think with all these features, we are ready to start building more complex tests such us maneuvers validation.

File: orekit/src/utils.py [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+        os.mkdir(dirname_path)
+    except FileExistsError:
+        pass
+
+
+def download_orekit_data(save_path=OREKIT_DATA_PATH, chunk_size=128):
+    """ Downloads orekit ZIP data """
+
+    # Prepare the logger
+    log = logging.getLogger(__name__)
```
Comment (astrojuanlu) [inline orekit/src/utils.py:29]: Sorry, I should have explained myself better! The instantiation should happen at module level. I'll push the change to your branch and merge.

**Review by astrojuanlu — APPROVED** [review]
This is good to go! 🚀

**Comment by astrojuanlu** [comment]
Merging! Thanks @jorgepiloto for taking this up and the patience 💪🏽

File: orekit/src/utils.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (astrojuanlu) [reply orekit/src/utils.py]: Fixed!

**Comment by astrojuanlu** [comment]
Oh, my bad... I now see that we merged `orekit-data.zip` here.

**Comment by astrojuanlu** [comment]
:warning: HISTORY OF THIS PULL REQUEST HAS BEEN REWRITTEN :warning:

**Pull request merged**